test/models/lookup_key_test.rb

```diff
@@ -184,36 +184,26 @@ def test_multiple_paths
       assert_equal val, param.default_value_before_type_cast
     end
   end
-
-  test "this is a smart variable?" do
-    assert lookup_keys(:two).class == VariableLookupKey
-  end
-
   test "this is a smart class parameter?" do
     assert_not_deprecated do
       assert lookup_keys(:complex).puppet?
     end
   end
 
   test "to_param should replace whitespace with underscore" do
-    lookup_key = VariableLookupKey.new(:key => "smart variable", :path => "hostgroup", :puppetclass => Puppetclass.first, :default_value => "default")
+    lookup_key = PuppetclassLookupKey.new(:key => "smart variable", :path => "hostgroup",
+                                          :default_value => "default")
     assert_equal "-smart_variable", lookup_key.to_param
   end
 
   test "should create smart variable with the same name as class parameters" do
```

Comment:
This looks like it's a test that can be removed.
We can't have 2 smart class parameters with the same name in the same class. If these are different classes that could be tested but it would require creating two entries.